salt/utils/minions.py

```diff
@@ -691,6 +691,13 @@ def _all_minions(self, expr=None):
                 mlist.append(fn_)
         return {"minions": mlist, "missing": []}
 
+    def syndic_minions(self):
+        syndic_cache_path = os.path.join(
+            self.opts["cachedir"], "syndics"
+        )
+        return [f for f in os.listdir(syndic_cache_path)]
```

Comment:
I need to get a list of syndics that are attached to this particular master. If there's a better way here, I'm all for it!

Reply:
Note: this is somewhat unreliable, in that the syndics aren't present until there's a `_syndic_return`

But it can easily be primed with a `salt \* test.ping` on the MoM

Reply:
Adding the timeout functionality and syndic job checking actually turns out to make the priming unnecessary. It's still probably a good idea to check this, but we also end out publishing the job and aggregating all of the syndics that return.